MyOnboarding/src/MyOnboardingApp.Services/Services/CreateItemService.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Threading.Tasks;
+using MyOnboardingApp.Contracts.Generators;
+using MyOnboardingApp.Contracts.Models;
+using MyOnboardingApp.Contracts.Repository;
+using MyOnboardingApp.Contracts.Services;
+using MyOnboardingApp.Contracts.Validation;
+
+namespace MyOnboardingApp.Services.Services
+{
+    internal class CreateItemService : AbstractUpsertService, ICreateItemService
+    {
+        private readonly ITodoListRepository _repository;
+        private readonly IIdGenerator<Guid> _idGenerator;
+        private readonly IDateTimeGenerator _dateTimeGenerator;
+
+
+        public CreateItemService(ITodoListRepository repository, IIdGenerator<Guid> idGenerator, IDateTimeGenerator dateTimeGenerator, IInvariantValidator<TodoListItem> validator)
+            : base(validator)
+        {
+            _repository = repository;
+            _idGenerator = idGenerator;
+            _dateTimeGenerator = dateTimeGenerator;
+        }
+
+
+        public async Task<IItemWithErrors<TodoListItem>> AddNewItemAsync(TodoListItem newItem)
+        {
+            var initializingData = GetGeneratedData();
+            return await TryCompleteAndStoreItemAsync(newItem, initializingData);
+        }
+
+
+        protected override async Task StoreToDatabase(TodoListItem completedItem)
+            => await _repository.AddNewItemAsync(completedItem);
+
+
+        private ItemInitializingData GetGeneratedData()
+        {
+            var id = _idGenerator.GetNewId();
+            var currentDateTime = _dateTimeGenerator.GetCurrentDateTime();
+
+            return new ItemInitializingData(id, currentDateTime, currentDateTime);
```

Comment:
There is a certain similarity (such as name and type of result, circumstances it is called in, ...) between this method and [the one in update service](https://github.com/petrareb/kentico-onboarding-cs/pull/4/files#diff-509d50ba5b162c08314cf315832c621bR48), so I would consider two things here:
1. wrapping
1. explicitly naming the last parameter (it is always a bit weird to pass the same variable twice, so one usually asks themselves "what is the other parameter about?")